### PR TITLE
Higher order component

### PR DESCRIPTION
**Pull request by andyOlarte514**
# Summary 
- Implementation of `higher order component`

# Trello Card
- https://trello.com/c/JkXViBb1

#ScreenShots
-
wolox app

### CODE REVIEW

File: training-react/tic-tac-toe/src/app/components/HOC/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react';
+
+import style from './styles.scss';
+
+function HigherOrderComponent(WrappedComponent) {
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/HOC/index.js]: Change the name the folder `HOC` for something more descriptive and also change the name of this component `HigherOrderComponent`

File: training-react/tic-tac-toe/src/app/components/HOC/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  class HOC extends Component {
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/HOC/index.js]: The same here and i think that you could change this to function component

File: training-react/tic-tac-toe/src/app/components/HOC/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    render() {
+      const { loading, ...props } = this.props;
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/HOC/index.js]: Remember the prop types

File: training-react/tic-tac-toe/src/app/components/HOC/styles.scss [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.loader {
+  display: inline-block;
+  height: 64px;
+  width: 64px;
+  &::after {
+    animation: lds-dual-ring 1.2s linear infinite;
+    border: 5px solid $white;
+    border-color: $white transparent $white transparent;
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/HOC/styles.scss]: The colors should be in `_colors.scss` file

File: training-react/tic-tac-toe/src/app/components/HOC/styles.scss [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    border-radius: 50%;
+    content: ' ';
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/HOC/styles.scss]: Delete the space

File: training-react/tic-tac-toe/src/app/components/HOC/styles.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  display: flex;
+  height: 100%;
+  justify-content: center;
+  position: absolute;
+  top: 0;
+  width: 100%;
+}
+
+.loader {
+  display: inline-block;
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/HOC/styles.scss]: I think that you don't need this

File: training-react/tic-tac-toe/src/app/components/RouteCheck/index.js [hunk 7]
```diff
@@ -1,41 +1,28 @@
...
-      <div>Loading...</div>
-    ) : (
-      <Router>
-        <Switch>
-          <ValidateRoute exact path={ROUTES.LOGIN} component={Login} />
-          <ValidateRoute isPrivate component={Home} />
-        </Switch>
-      </Router>
-    );
+    const loading = this.props.loading;
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/components/RouteCheck/index.js]: Use destructuring here

File: training-react/tic-tac-toe/src/app/screens/Home/screens/Profile/index.js [hunk 8]
```diff
@@ -16,11 +16,11 @@ class Profile extends Component {
   }
 
   render() {
-    const { onUpdateUser, isLoadedProfile, infoUser, updateUserError, isSuccessUpdate } = this.props;
+    const { onUpdateUser, loading, infoUser, updateUserError, isSuccessUpdate } = this.props;
     return (
       <Layout
         onSubmit={onUpdateUser}
-        isLoaded={isLoadedProfile}
+        loading={loading}
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/screens/Home/screens/Profile/index.js:23]: You don't use this prop in the `Layout` component so delete it

File: training-react/tic-tac-toe/src/app/screens/Login/index.js [hunk 9]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { connect } from 'react-redux';
 import actionCreators from '@redux/login/actions';
 import PropTypes from 'prop-types';
+import HCO from '@components/HOC';
```
Comment (henryzarza) [inline training-react/tic-tac-toe/src/app/screens/Login/index.js]: Change this name

**Review by henryzarza — CHANGES_REQUESTED** [review]

**Review by henryzarza — APPROVED** [review]

**Review by wfolini — COMMENTED** [review]
Great! Looks good to me. Can u upload a GIF showing the HOC in action?

**Review by wfolini — APPROVED** [review]
:clap:

**Pull request merged**